app/models/admin/tag_filter.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class Admin::TagFilter
+  KEYS = %i(
+    status
+    name
+    order
+  ).freeze
+
+  attr_reader :params
+
+  def initialize(params)
+    @params = params.to_h.symbolize_keys
+  end
+
+  def results
+    scope = Tag.reorder(nil)
+
+    params.each do |key, value|
+      next if key == :page
+
+      scope.merge!(scope_for(key, value)) if value.present?
+    end
+
+    scope
+  end
+
+  private
+
+  def scope_for(key, value)
+    case key
+    when :status
+      status_scope(value)
+    when :name
+      Tag.search_for(value.to_s.strip, params[:limit], params[:offset], exclude_unlistable: false)
+    when :order
+      order_scope(value)
+    else
+      raise Mastodon::InvalidParameterError, "Unknown filter: #{key}"
+    end
+  end
+
+  def status_scope(value)
+    case value.to_s
```

Comment:
@oneiros can the rails params return a value that's **not** a string?

Reply:
Not in case of "regular" params sent by a browser. Since those are not typed, they will always be strings.

There is a special case, where a client can send JSON data and they end up in `params` as well. Since they are parsed as JSON, other data types may appear.

Of course this is not the case here. But there is another consideration: The controller could set a value before passing the params in here. I think here it already does to set a default order. One might want to allow symbols in this case, because they might look nicer in the controller. In that case `#to_s` would make sense here.

I would leave this up to you if you would like to allow this / guard against this here.